### DISCUSSION TITLE
Output directory different for command line build vs. Visual Studio

### DISCUSSION THREAD

**Discussion by george7378** [wixtoolset/issues#7510] [Questions] [answered]
I'm trying to get a WiX v4 project to build on a build server, and running into a problem with the build output directories that I can't get my head around. The project is set up so that the code compiles first, then I have my first .wixproj which uses Heat to harvest the files from the C# build directory and put them into an MSI. Then I have a second .wixproj which creates a bundle, first installing the .net runtime if needed and then installing the MSI I just created.

Everything works fine when I build using visual studio - the first MSI is placed in the bin\x86\Release\ directory, where it is found by the bundle like this:

`<MsiPackage SourceFile="..\MyInstaller\bin\x86\Release\MyInstaller.msi" bal:DisplayInternalUICondition="1" />`

However when I build the same project using _dotnet build_ from the command line (or when the build server does the same thing) the build fails because the MsiPackage can't find the source file.

When I look, it's because the MSI is now placed in 'bin\Release' rather than 'bin\x86\Release'. The same happens when I use msbuild from the command line.

Any idea why the directory is changing?

**Comment by jcasale** [comment, marked as answer]
When you add projects as dependencies to a WIX project in Visual Studio, you can reference their output directly without a path.

That may also work with MSI outputs from a WIX project. Try adding the first WIX project to the second one as a dependency and reference the MSI as follows:

```XML
<MsiPackage SourceFile="MyInstaller.msi" bal:DisplayInternalUICondition="1" />
```

**Reply by george7378** [reply]
Ah yes, that did it - thanks for the info!

The original problem (different directories) still exists though - not sure if anyone wants to investigate it as a bug or maybe knows the reason why it happens?

**Reply by jcasale** [reply]
Great!

The directories changing is an msbuild nuance.

You can explicitly specify the base output and base intermediate path in the csproj, or leave it up to msbuild. I don't know why it differs sometimes when you leave it blank.
